src/functional/iterator/transforms/eta_reduction.py

```diff
@@ -5,7 +5,11 @@
 class EtaReduction(NodeTranslator):
     """Eta reduction: simplifies `λ(args...) → f(args...)` to `f`."""
 
-    def visit_Lambda(self, node: ir.Lambda) -> ir.Node:
+    def visit_Lambda(self, node: ir.Lambda, **kwargs) -> ir.Node:
+        """Do not apply eta "inside" reductions."""
+        if "in_reduce" in kwargs and kwargs["in_reduce"]:
```

Comment:
This is the first work around mentioned in the limitations: "The eta pass is disabled for the body of reductions. This is because the eta pass generates an explicit call to plus for cases where the body of the reduction is not a binary operation, but a simple field access"

I think we agreed that this is a bug in the eta pass, and to handle this in a future clean up effort regarding itir or that passes.

Reply:
Thanks, for reminding me. Maybe we should fix it in the ir -> gtfn_ir, by wrapping unapplied operators back into a lambda. Also because
```
reduce(plus,0)(...)
```
is legal in iterator IR, we are just lucky that the frontend doesn't generate that.

Reply:
Ok, fair enough. 
* If this is legal `itir` then this is an untested situation (since I added the gtfn_im backend to the back ends being tested and it passes). No criticism, there are infinitely many programs and we can't test them all. Just noting that we should add this as a test after this is fixed.
* Do you feel fixing this is a precondition to this PR?

Reply:
I'd say we shouldn't add the special case in eta reduction, which then probably makes it a precondition?

Reply:
I can try to implement it quickly.